src/core/systemtrayicon.cpp

```diff
@@ -47,31 +49,36 @@ SystemTrayIcon::SystemTrayIcon(QObject *parent)
 
 QPixmap SystemTrayIcon::CreateIcon(const QPixmap &icon, const QPixmap &grey_icon) {
 
-  Q_UNUSED(grey_icon);
-
   QRect rect(icon.rect());
 
-  // The angle of the line that's used to cover the icon.
-  // Centered on rect.topRight()
-  double angle = double(100 - song_progress()) / 100.0 * M_PI_2 + M_PI;
-  double length = sqrt(pow(rect.width(), 2.0) + pow(rect.height(), 2.0));
+  QSettings s;
+  s.beginGroup(AppearanceSettingsPage::kSettingsGroup);
+  bool trayicon_progress = s.value(AppearanceSettingsPage::kTrayIconProgress, false).toBool();
```

Comment:
Move this to MainWindow::ReloadSettings() so we don't reload the setting each time the icon is painted.
Then add a public method in SystemTrayIcon to call when the setting is reloaded.